lib/components/viewers/stop-viewer.js

```diff
@@ -104,13 +104,22 @@ class StopViewer extends Component {
       (s) => s.id === this.props.stopData.id
     ) !== -1
 
-  componentDidUpdate(prevProps) {
+  componentDidUpdate(prevProps, prevState) {
     const { fetchStopInfo, viewedStop } = this.props
+    const { date, isShowingSchedule } = this.state
 
     if (prevProps.viewedStop?.stopId !== viewedStop?.stopId) {
       // Reset state to default if stop changes (i.e., show next arrivals).
       this.setState(defaultState)
       fetchStopInfo(viewedStop)
+    } else if (
+      isShowingSchedule &&
+      isShowingSchedule !== prevState.isShowingSchedule
```

Comment:
Would it maybe be more clear to write 
`isShowingSchedule && !prevState.isShowingSchedule`?